dvc/repo/metrics/show.py

```diff
@@ -226,22 +222,30 @@ def _read_metrics(self, metrics, branch):
         if not typ:
             typ = os.path.splitext(out.path.lower())[1].replace(".", "")
         if out.use_cache:
-            metric = _read_metrics_filesystem(
-                self.cache.local.get(out.checksum),
-                typ=typ,
-                xpath=xpath,
-                rel_path=out.rel_path,
-                branch=branch,
-            )
+            open_fun = open
+            path = repo.cache.local.get(out.checksum)
```

Comment:
But in this case `Metrics file '{}' does not exist at branch: '{}'` doesn't really fit, since path is a cache file for metric file :slightly_smiling_face:

Reply:
Not exactly, we pass out.relpath, which is "common part" for both use cases. We always get relative path of our out.

Reply:
Ah, right. Thanks!